chapter4.adoc

```diff
@@ -221,28 +221,67 @@ follows:
 ], config:{lanes: 1, hspace:1024}}
 ....
 
-=== Caching
-
-Implementations with virtual memory are permitted to cache translations and
-permissions in address translation cache structures. Similarly, access
-permissions from the `MTT` lookup may be cached. The `PMP` and `MTT` settings
-for the resulting physical address may be checked (and possibly cached) at any
-point between the address translation and the explicit memory access. If
-caching is occuring, when the `MTT` settings are modified, `M-mode` software
-must synchronize the cached `MTT` state with the virtual memory system and any
-`PMP`, `MTT` or address-translation caches. This is accomplished by executing
-an `SFENCE.VMA` instruction with `rs1=x0` and `rs2=x0`, or `HFENCE.GVMA` as
-needed, after the `MTT` is modified. If page-based virtual memory is not
-implemented, memory accesses check the `PMP` settings synchronously, but may
-check `MTT` settings that are cached, so a `MTT` invalidation (`MTTINVAL`)
-instruction is needed. When Svinval is implemented, `MTTINVAL` is only ordered
-against `SFENCE.W.INVAL` and `SFENCE.INVAL.IR` instructions. As part of the
-`MTT` update, the RDSM must ensure that it uses `SFENCE.W.INVAL` to guarantee
-that any previous stores to `MTT` are made visible before invoking the
-`MTTINVAL`. The RDSM must then use `SFENCE.INVAL.IR` to guarantee that all
-subsequent implicit references to `MTT` are ordered to be after the `MTT` cache
-invalidation.
-
-_[TBD - register interface for flushing all MTT cached entries, vs specific
-physical address at page size granularity]._
-
+=== Access Enforcement and Fault Reporting
+
+As shown in <<mtt-lookup>>, MTT lookup composes with, but does not require,
+page-based virtual memory (MMU, IOMMU) and physical memory protection mechanisms
+(PMP, Smepmp, IOPMP). When paging is enabled, instructions that access virtual
+memory may result in multiple physical-memory accesses, including (implicit
+S-mode) accesses to the page tables. MTT checks also apply to these implicit
+S-mode accesses - those accesses will be treated as reads for translation and as
+writes when A/D bits are updated in page table entries when `Svadu` is
+implemented.
+
+MTT is checked for all accesses to physical memory, including accesses that have
```

Comment:
Suggestion: MTT is checked for all accesses to physical memory **unless the effective privilege mode is M,** including accesses that have....

(The disclaimer to effective privilege mode M is already done in chapter 3, but some redundancy does not help, as this section is more precise.)

Also in the same context: the privileged spec in section 3.7 is more explicit for PMP checks: _"...PMP checks are applied to .... data accesses in any mode when the MPRV bit in mstatus is set and the MPP field in mstatus contains S or U."_
Should we add an informative sentence to state the same for MTT? This may not be obvious to readers.

Reply:
yes i will add that clarifications.